server/src/test/java/org/elasticsearch/http/AbstractHttpServerTransportTests.java

```diff
@@ -173,6 +177,63 @@ public HttpStats stats() {
         }
     }
 
+    public void testIncorrectHeaderHandling() {
+
+        final ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        try (AbstractHttpServerTransport transport =
+                 new AbstractHttpServerTransport(Settings.EMPTY, networkService, bigArrays, threadPool, xContentRegistry(),
+                     new HttpServerTransport.Dispatcher() {
+                         @Override
+                         public void dispatchRequest(RestRequest request, RestChannel channel, ThreadContext threadContext) {
+                             Assert.fail();
+                         }
+
+                         @Override
+                         public void dispatchBadRequest(RestChannel channel, ThreadContext threadContext, Throwable cause) {
+                             assertThat(cause, instanceOf(RestRequest.MediaTypeHeaderException.class));
```

Comment:
Is there anywhere that we can assert that the header name at fault is included in the error sent back to the user ?

Reply:
great you noticed this. in fact the header name was not surfaced on a response.
I refactored this to return something like 
```
{
  "error": {
    "root_cause": [
      {
        "type": "media_type_header_exception",
        "reason": "Invalid media-type value on header [Accept]"
      }
    ],
    "type": "media_type_header_exception",
    "reason": "Invalid media-type value on header [Accept]",
    "caused_by": {
      "type": "illegal_argument_exception",
      "reason": "invalid media type [texffff]"
    }
  },
  "status": 400
}
```